network-api/networkapi/buyersguide/views.py

```diff
@@ -29,10 +29,12 @@
 locale_regex = re.compile(r"^/[a-z]{2}(-[A-Z]{2})?/")
 
 
-if settings.USE_CLOUDINARY:
-    MEDIA_URL = settings.CLOUDINARY_URL
-else:
-    MEDIA_URL = settings.MEDIA_URL
+def get_media_url(use_cloudinary=False):
+    if use_cloudinary:
+        return settings.CLOUDINARY_URL
+    else:
+        aws_location_exists = hasattr(settings, "AWS_LOCATION") and settings.AWS_LOCATION
+        return f"{settings.MEDIA_URL}{settings.AWS_LOCATION}/" if aws_location_exists else settings.MEDIA_URL
```

Comment:
the `hasattr` check will always be true, because it's either "what's in the env file" or an empty string, so you'll want to do this as:

```suggestion
def get_media_url(use_cloudinary=False):
    if use_cloudinary:
        return settings.CLOUDINARY_URL
    elif settings.AWS_LOCATION:
        return f"{settings.MEDIA_URL}{settings.AWS_LOCATION}/"
    return  settings.MEDIA_URL
```

(no `else` because all steps are `return`ing)

Reply:
I commited your suggested code, @Pomax . But I do have questions around it --

Do we always have to refer to what’s defined in `settings.py` when we do these checks? If I only care about `AWS_LOCATION` for the category view I should only need to check the existence and value of it and not other variables? like… why should category view care about whether S3 is used or not?

Reply:
Convo happened on Slack. Marking this as resolved.